test/mcmc/gibbs.jl

```diff
@@ -135,10 +281,271 @@ using Turing.RandomMeasures: ChineseRestaurantProcess, DirichletProcess
                 m[k] ~ Normal(1.0, 1.0)
             end
         end
-        model = imm(randn(100), 1.0)
+        model = imm(Random.randn(100), 1.0)
         # https://github.com/TuringLang/Turing.jl/issues/1725
-        # sample(model, Gibbs(MH(:z), HMC(0.01, 4, :m)), 100);
-        sample(model, Gibbs(PG(10, :z), HMC(0.01, 4, :m; adtype=adbackend)), 100)
+        # sample(model, Gibbs(; z=MH(), m=HMC(0.01, 4)), 100);
+        sample(model, Gibbs(; z=PG(10), m=HMC(0.01, 4; adtype=adbackend)), 100)
+    end
+
+    @testset "dynamic model with dot tilde" begin
+        @model function dynamic_model_with_dot_tilde(num_zs=10)
+            z = Vector(undef, num_zs)
+            z .~ Exponential(1.0)
+            num_ms = Int(round(sum(z)))
+            m = Vector(undef, num_ms)
+            return m .~ Normal(1.0, 1.0)
+        end
+        model = dynamic_model_with_dot_tilde()
+        # TODO(mhauru) This is broken because of
+        # https://github.com/TuringLang/DynamicPPL.jl/issues/700.
+        @test_broken (
+            sample(
+                model,
+                Gibbs(; z=NUTS(; adtype=adbackend), m=HMC(0.01, 4; adtype=adbackend)),
```

Comment:
If you do discrete `z`, then `NUTS` won't work here (but it also won't do the correct thing when you do continuous + `round`, though it'll probably still run)

Reply:
The `round` ruins gradient info

Reply:
Now using Poisson and PG